packages/eslint-plugin/src/rules/sort-type-union-intersection-members.ts

```diff
@@ -32,12 +32,21 @@ function getGroup(node: TSESTree.TypeNode): Group {
     case AST_NODE_TYPES.TSIntersectionType:
       return Group.intersection;
 
+    case AST_NODE_TYPES.TSAbstractKeyword:
     case AST_NODE_TYPES.TSAnyKeyword:
+    case AST_NODE_TYPES.TSAsyncKeyword:
     case AST_NODE_TYPES.TSBigIntKeyword:
     case AST_NODE_TYPES.TSBooleanKeyword:
+    case AST_NODE_TYPES.TSDeclareKeyword:
+    case AST_NODE_TYPES.TSExportKeyword:
     case AST_NODE_TYPES.TSNeverKeyword:
     case AST_NODE_TYPES.TSNumberKeyword:
     case AST_NODE_TYPES.TSObjectKeyword:
+    case AST_NODE_TYPES.TSPrivateKeyword:
+    case AST_NODE_TYPES.TSProtectedKeyword:
+    case AST_NODE_TYPES.TSPublicKeyword:
+    case AST_NODE_TYPES.TSReadonlyKeyword:
+    case AST_NODE_TYPES.TSStaticKeyword:
```

Comment:
none of these keywords are ever valid in a union or intersection.
they should instead be placed in the last block which throws an error

Reply:
@bradzacher Only the ones I added or all of them?